lib/extend/filter.js

```diff
@@ -62,6 +62,7 @@ class Filter {
     const ctx = options.context;
     const args = options.args || [];
 
+    args.push(data);
     args.unshift(data);
```

Comment:
```suggestion
    args.push(data);
    args.unshift(data);
```

If we already `unshift` the data as the first element of the `args`, why `push` it again?

Reply:
Almost all filters cb will change the first elements of args, that means maybe one filter's callback may change an object element to a string, and the next callback can only get the string as "Function parameters", and lose other object info.

For example:  when use `ctx.execFilterSync('post_permalink', this, {context: ctx});`  trigger `post_permalink` cb(`this` point to a `Post Object`). The first filter callback register as `hexo.extend.filter.register('post_permalink', function(data) { return data.title });`  

And then the second filter callback can only get `post title` as the only "Function parameters", and it lost other info about the `Post Object`.

```js
// example1
hexo.extend.filter.register('post_permalink', function(data){
  return data;
});
```

In example1, the `filter1` parameter data is  `post title` and lose `Post Object` other properties.


**All of args content will be passed to filter's next callback as arguments.**


When push data to `args` again, This will as filter's callback in last parameter, and the second can still get the `Post Object` .

```js
// example2
hexo.extend.filter.register('post_permalink', function(data, post){
// do sth
});
```
In example2, the `post` parameter is from  `args.push(data);` 

So i use `args.push(data)` hope give other filter's callback more infomation to do sth.